tensornetwork/backends/shell/shell_backend.py

```diff
@@ -255,3 +255,13 @@ def multiply(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
     a = np.ones(tensor1.shape)
     b = np.ones(tensor2.shape)
     return ShellTensor((a * b).shape)
+
+  def add(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
+    a = np.ones(tensor1.shape)
+    b = np.ones(tensor2.shape)
+    return ShellTensor((a + b).shape)
+
+  def sub(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
+    a = np.ones(tensor1.shape)
+    b = np.ones(tensor2.shape)

```

Comment:
Not sure I agree with how this is implemented. This will blow up if the shapes are huge (which would be the main reason to use the shell backend in the first place).

Reply:
I agree, just check if the shapes  the tensors match and then create a ShellTensor directly

Reply:
Changed as suggested. Added another unit test to check mismatch shape. Please take another look.